packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -16,36 +17,50 @@ import type {
   SingleEventHandler,
 } from './eventTypes'
 
+export type DomainEventEmitterDependencies<SupportedEvents extends CommonEventDefinition[]> = {
+  eventRegistry: EventRegistry<SupportedEvents>
+  metadataFiller: MetadataFiller
+  // TODO: make them mandatory is a breaking change, decide if we are fine with that
```

Comment:
TBD

Reply:
After thinking about it, I think that we make logger and error-reporter mandatory, as it is really useful to understand if a BG listener failed